TFNetworkLayer.py

```diff
@@ -1045,16 +1047,34 @@ def batch_norm(self, data,
       https://github.com/deepmind/sonnet/blob/master/sonnet/python/modules/batch_norm.py
     """
     with tf.variable_scope("batch_norm"):
+      import numpy as np
       if masked_time:
+        assert not use_frequency
+        # Default case
         x = data.get_placeholder_flattened(keep_dims=True)
         mean, variance = tf.nn.moments(x, axes=[0], keep_dims=True)
+        variables_shape = data.get_bc_spatial_batch_shape()
+        squeeze_variables = False
       else:
+        # Variables have to be squeezed in order to save/load the variables in
+        # different data formatrs (NCHW/NHWC) => set squeeze_variables
+        squeeze_variables = True
         x = data.placeholder
-        mean, variance = tf.nn.moments(x, axes=data.get_axes(exclude_feature=True), keep_dims=True)
+        axes = data.get_axes(exclude_feature=True)
+        if use_frequency:
+          opts = {data.feature_dim_axis: -1, data.get_axis_from_description("s:1"): -1}
+          axes.remove(data.get_axis_from_description('s:1'))
+        else:
+          opts = {}
+        default_shape = data.get_bc_shape(opts)
```

Comment:
Don't call this `default_shape`. Why `default`? Call this `data_bc_shape` or so.